### PR TITLE
Add health checker for nsx operator

### PR DESCRIPTION
**Pull request by seanpang-vmware**

### CODE REVIEW

File: pkg/nsx/client.go [hunk 1]
```diff
@@ -41,12 +51,21 @@ func GetClient(cf *config.NSXOperatorConfig) *Client {
 	groupClient := domains.NewGroupsClient(restConnector(cluster))
 	securityClient := domains.NewSecurityPoliciesClient(restConnector(cluster))
 	ruleClient := security_policies.NewRulesClient(restConnector(cluster))
+	nsxChecker := func(req *http.Request) error {
+		health := cluster.Health()
+		if GREEN == health {
```
Comment (dantingl) [inline pkg/nsx/client.go:56]: orange should also be fine?

File: pkg/nsx/client.go [hunk 2]
```diff
@@ -12,18 +14,26 @@ import (
...
 
 	"github.com/vmware-tanzu/nsx-operator/pkg/config"
 	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/ratelimiter"
 )
 
+var typeToString = map[ClusterHealth]string{
+	RED:    "RED",
+	ORANGE: "ORANGE",
+	GREEN:  "GREEN",
+}
```
Comment (dantingl) [inline pkg/nsx/client.go:27]: I think we could define constants together for cluster.go and client.go at the same place

File: pkg/nsx/client.go [hunk 3]
```diff
@@ -41,12 +51,21 @@ func GetClient(cf *config.NSXOperatorConfig) *Client {
 	groupClient := domains.NewGroupsClient(restConnector(cluster))
 	securityClient := domains.NewSecurityPoliciesClient(restConnector(cluster))
 	ruleClient := security_policies.NewRulesClient(restConnector(cluster))
+	nsxChecker := func(req *http.Request) error {
```
Comment (dantingl) [inline pkg/nsx/client.go:54]: Could we define a separate function in case that we may add more logic for health check later

**Comment by seanpang-vmware** [comment]
can not update this request due to repo change to public, will generate a new pull request.

**Pull request closed without merging**